File: extractor.js
const fs = require('fs');
const content = fs.readFileSync('/Users/macbook/site gelatina moujaro/tmp_script.js', 'utf8');

const regex = /ingredients:\[(.*?)\]/g;
let match;
let count = 0;
while ((match = regex.exec(content)) !== null) {
    count++;
}
console.log("Total ingredients arrays found:", count);

// Try to find the structure of one recipe
const recipeRegex = /\{title:"([^"]+)",(?:image:"[^"]*",)?ingredients:\[(.*?)\],instructions:"([^"]+)"\}/g;
const recipes = [];
let r;
while ((r = recipeRegex.exec(content)) !== null) {
    recipes.push({
        title: r[1],
        ingredients: r[2],
        instructions: r[3]
    });
}
console.log("Total fully matched recipes:", recipes.length);

if (recipes.length > 0) {
    console.log("Sample:", JSON.stringify(recipes[0]));
    fs.writeFileSync('/Users/macbook/site gelatina moujaro/extracted_recipes.json', JSON.stringify(recipes, null, 2));
} else {
    // Let's get a loose match around "ingredients"
    const loose = content.match(/.{0,80}ingredients:\[.{0,100}/);
    if (loose) console.log("Loose match:", loose[0]);
}

// Let's also look for id:, title:, items:
const categories = content.match(/\{id:"[^"]+",title:"[^"]+",items:\[(.*?)\]\}/g);
if (categories) {
    console.log("Found categories:", categories.length);
}

File: extractor.py
import re
import json

with open('/Users/macbook/site gelatina moujaro/tmp_script.js', 'r', encoding='utf-8') as f:
    content = f.read()

# Look for all occurrences of ingredients array pattern
ingredients_matches = re.finditer(r'ingredients:\[(.*?)\]', content)
print(f"Total ingredients arrays found: {sum(1 for _ in ingredients_matches)}")

# Try to find objects that have 'title' or 'nome', 'ingredients', and 'instructions'
# The data might be like: {title:"Omelete",ingredients:["Ovo","Sal"],instructions:"Misture e frite"}
recipe_regex = r'\{[^{]*?title:"([^"]+)"[^{]*?ingredients:\[(.*?)\][^{]*?\}'
recipes = []

for match in re.finditer(recipe_regex, content):
    recipes.append({
        "title": match.group(1),
        "ingredients": match.group(2)
    })

print(f"Total fully matched recipes: {len(recipes)}")

if recipes:
    print("Sample:", recipes[0])
    with open('/Users/macbook/site gelatina moujaro/extracted_recipes.json', 'w', encoding='utf-8') as f:
        json.dump(recipes, f, ensure_ascii=False, indent=2)
else:
    loose_match = re.search(r'.{0,80}ingredients:\[.{0,100}', content)
    if loose_match:
        print("Loose match around ingredients:")
        print(loose_match.group(0))

# Try looking for "categories"
categories_regex = r'id:"([^"]+)",title:"([^"]+)",items:\[(.*?)\]'
cats = list(re.finditer(categories_regex, content))
print(f"Found categories strings: {len(cats)}")
if cats:
    print("Category sample:",Cats[0].group(2))
